src/software/backend/simulator_backend.cpp

```diff
@@ -28,22 +28,49 @@ void SimulatorBackend::setSimulationSpeed(
     this->simulation_speed_mode = simulation_speed_mode;
 }
 
-bool SimulatorBackend::runSimulation(World world, const Duration &timeout)
+bool SimulatorBackend::runSimulation(
+    const std::vector<ValidationFunction> &validation_functions, World world,
```

Comment:
I'm not a fan of doing the validation like this. I would much prefer that we get a world from the simulator and run validation on that instead. This seems like unnecessary coupling.

Reply:
You want to make the test writer handle checking the validation?